token-swap/program/src/curve.rs

```diff
@@ -27,14 +27,19 @@ impl SwapResult {
         fee_denominator: u64,
     ) -> Option<SwapResult> {
         let invariant = swap_source_amount.checked_mul(swap_destination_amount)?;
-        let new_source_amount = swap_source_amount.checked_add(source_amount)?;
-        let new_destination_amount = invariant.checked_div(new_source_amount)?;
-        let remove = swap_destination_amount.checked_sub(new_destination_amount)?;
-        let fee = remove
+
+        // assess the fee to calculate the amount swapped
```

Comment:
```suggestion
        // debit the fee to calculate the amount swapped
```
Assess is correct, of course, but its other meaning makes it ambiguous